### ISSUE TITLE
Branch rules are not applies when the initial push to origin is done.

### ISSUE THREAD

**Issue by KieranDevvs** [isaacs/github#1391]
As the title states, branch rules are ignored when the branch is first created on remote. This kind of defeats the purpose of the rules. Our current policy is to enforce that all work has to be merged into a feature branch that is protected. This works fine unless the feature branch is new, as this allows someone to make commits to that branch before pushing it to remote. 

We use this policy to keep pull requests small. Users initially create a feature branch that is protected by the necessary matching branch rule. Users then create personal branches and then create pull requests to merge them into the feature branch. Once the feature is complete, the entire feature does not need to be re-reviewed as subsequent reviews have already covered the entire feature.

As stated before, this process breaks down if commits are allowed to be pushed to a protected branch that has not already been created yet.
